morphir/src/org/finos/morphir/util/PrintMDMJson.scala

```diff
@@ -0,0 +1,146 @@
+package org.finos.morphir.util
+
+import _root_.org.finos.morphir.naming.*
+import _root_.org.finos.morphir.datamodel.Label
+import _root_.org.finos.morphir.datamodel.EnumLabel
+import _root_.org.finos.morphir.datamodel.Data
+import _root_.org.finos.morphir.datamodel.Concept
+import org.finos.morphir.datamodel.Data.{Optional, Result}
+import zio.json.ast._
+import zio.json.EncoderOps
+
+object PrintMDMJson {
+
+  def prettyPrint(
+      any: Any,
+      detailLevel: DetailLevel = DetailLevel.BirdsEye,
+      fieldNames: FieldNames = FieldNames.Hide
+  ): String =
+    toJson(any, detailLevel, fieldNames).toJsonPretty
+
+  def compactPrint(
+      any: Any,
+      detailLevel: DetailLevel = DetailLevel.BirdsEye,
+      fieldNames: FieldNames = FieldNames.Hide
+  ): String =
+    toJson(any, detailLevel, fieldNames).toJson
+
+  private def toJson(any: Any, detailLevel: DetailLevel, fieldNames: FieldNames): Json =
+    any match {
+      case name: Name   => Json.Str(name.toTitleCase)
+      case qn: FQName   => Json.Str(if (detailLevel.hideFQNames) qn.localName.toTitleCase else qn.toStringTitleCase)
+      case label: Label => Json.Str(label.value)
+      case v: Data      => dataToJson(v)
+      case v: Concept   => conceptToJson(v, detailLevel)
+      case other        => Json.Str(other.toString)
+    }
+
+  private def dataToJson(data: Data): Json =
+    data match {
+      case v: Data.Basic[?] => Json.Str(v.toString)
+      case v: Data.Case => Json.Obj(
+          "enumLabel" -> Json.Str(v.enumLabel),
+          "values" -> Json.Arr(v.values.map {
+            case (enumLabel, data) => Json.Obj(enumLabel.toString -> dataToJson(data))
+          }*)
+        )
+      case v: Data.Tuple  => Json.Arr(v.values.map(dataToJson)*)
+      case v: Data.Struct => Json.Obj(v.values.map { case (k, v) => k.value -> dataToJson(v) }*)
+      case v: Data.Record => Json.Obj(v.values.map { case (k, v) => k.value -> dataToJson(v) }*)
+      case v: Data.Optional => v match {
+          case Optional.Some(data, _) => dataToJson(data)
+          case Optional.None(_)       => Json.Null
+        }
+      case v: Data.Result => v match {
+          case Result.Ok(data, _)  => dataToJson(data)
+          case Result.Err(data, _) => dataToJson(data)
+        }
+      case v: Data.List => Json.Arr(v.values.map(dataToJson)*)
+      case v: Data.Map => Json.Obj(v.values.map {
+          case (k, v) => k.toString -> dataToJson(v)
+        }.toList*)
+      case v: Data.Set     => Json.Arr(v.values.toList.map(dataToJson)*)
+      case v: Data.Union   => dataToJson(v.value)
+      case v: Data.Aliased => dataToJson(v.data)
+
+    }
+
+  private def conceptToJson(concept: Concept, detailLevel: DetailLevel): Json =
```

Comment:
The same comment as above for `dataToJson` and recursion applies to `conceptToJson`.